code/controllers/_DynamicAreaLighting_TG.dm

```diff
@@ -209,13 +209,23 @@ atom/proc/AddLuminosity(delta_luminosity)
 //This slightly modifies human luminosity. Source of light do NOT stack.
 //When you drop a light source it should keep a running total of your actual luminosity and set it accordingly.
 mob/SetLuminosity(new_luminosity, trueLum)
-	if(!new_luminosity) //Our new addition is positive. Add it to our running total.
+	luminosity_total = 0 //reset this
+	var/current_light
+	for(var/L in light_sources) //get the most powerful light source
+		//to_chat(world, "SET LUMINOSITY MOB DEBUG L: [L] current_light: [current_light]")
+		if(current_light > L)
+			continue
+		current_light = L
+
+	//to_chat(world, "SET LUMINOSITY MOB DEBUG 2 current_light: [current_light]")
+	if(!current_light) //Our new addition is positive. Add it to our running total.
 		return..(0, trueLum)  //Set to ZERO.
 	else
-		luminosity_total += new_luminosity //Keep track of our new total.
+		luminosity_total += current_light //Keep track of our new total.
```

Comment:
Since we are resetting this at the start of the proc, there's no need to have the variable `current_light`, nor these `if` and `else right here.

But the whole of this logic has to change, else this will break the Boiler's luminosity, or any mob which is luminous by itself without the aid of any items.

First you need to take into account if a `new_luminosity` value is passed. If it is, someone is trying to add a source of light. Ideally there are no more than one of those per mob.